pkgs/tools/misc/dust/default.nix

```
@@ -2,19 +2,21 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "dust";
-  version = "0.4.4";
+  version = "0.5.0";
 
   src = fetchFromGitHub {
     owner = "bootandy";
     repo = "dust";
     rev = "v${version}";
-    sha256 = "1qbh9vgdh0xmh4c78fm0rd1sgb01n656p3cr4my7ymsy81ypx9y7";
+    sha256 = "1csv1wiag4rdjxsggsxvi4q1krbg9153pl6aaq1cpfh32x4l12lv";
+    # Remove unicode file names which leads to different checksums on HFS+
+    # vs. other filesystems because of unicode normalisation.
+    extraPostFetch = ''
+      rm -r $out/src/test_dir3/
```

```suggestion
      rm -rf $out/src/test_dir3/
```
Unfortunately the `extraPostFetch` has the property of not being idempotent when round-tripped through a hashed-mirror.

In more elaborate detail: the hash `1csv1wiag4rdjxsggsxvi4q1krbg9153pl6aaq1cpfh32x4l12lv` here is computed over the result of the cleaned git clone with the post fetch run, which means it's a hash that doesn't depend this test_dir3, as you desire; however, if nix finds something in a hashed mirror at has `1csv1wiag4rdjxsggsxvi4q1krbg9153pl6aaq1cpfh32x4l12lv`, it will download it and then *still* run the `extraPostFetch`, because it's a post fetch hook that ought to be run after fetching, of course.  That's not a problem if the `extraPostFetch` is idempotent, which we can do by adding the `f` flag.

---

We could also do something like this, which avoids the need for the `extraPostFetch` entirely.

```nix
sha256 = if stdenv.isDarwin then "xxx" else "yyy";
```

---

Thanks, the detail about `extraPostFetch` is good to know